### PR TITLE
147 add sha256 external function

### PR DESCRIPTION
**Pull request by Ben-Rey**
I had difficulties to retrieve the value returned by the functions used in vm-mock. It would be great if someone could tell me how to do it.

### CODE REVIEW

**Comment by Thykof** [comment]
After some tries I finally manage to make it work, don't hesitate to ask questions :)

**Review by Thykof — APPROVED** [review]

File: assembly/env/env.ts [hunk 1]
```diff
@@ -181,4 +181,7 @@ export namespace env {
 
   @external("massa", "assembly_script_get_op_data")
   export declare function getOpData(key: StaticArray<u8>): StaticArray<u8>;
+
+  @external("massa", "assembly_script_sha256")
```
Comment (aoudiamoncef) [inline assembly/env/env.ts:185]: We changed the design of this function to : `assembly_script_hash_sha256`

File: vm-mock/vm.js [hunk 2]
```diff
@@ -265,6 +283,12 @@ export default function createMockedABI(memory, createImports, instantiate, bina
         return true;
       },
 
+      assembly_script_sha256(aPtr) {
```
Comment (aoudiamoncef) [inline vm-mock/vm.js:286]: same

**Review by Thykof — APPROVED** [review]

File: vm-mock/vm.js [hunk 3]
```diff
@@ -265,6 +283,12 @@ export default function createMockedABI(memory, createImports, instantiate, bina
         return true;
       },
 
+      assembly_script_hash_sha256(aPtr) {
+        const stringToHash = byteArrToUTF8String(getArrayBuffer(aPtr));
```
Comment (peterjah) [inline vm-mock/vm.js:287]: stringToHash is not well named !

Comment (Ben-Rey) [reply vm-mock/vm.js:287]: Hi Pierre, thanks for the review. Is decodedString better ?

File: assembly/__tests__/vm-mock.spec.ts [hunk 4]
```diff
@@ -64,4 +64,11 @@ describe('Testing mocked Storage and CallStack', () => {
   test('Testing event', () => {
     generateEvent("I'm an event ");
   });
+
+  test('Testing sha256', () => {
+    const result = bytesToString(sha256(stringToBytes('something')));
+    expect(result).toBe(
+      '3fc9b689459d738f8c88a3a48aa9e33542016b7a4052e001aaa536fca74813cb',
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:71]: Are you sure that the blockchain is using the same case ? '3f...' and not '3F...'

**Review by gregLibert — APPROVED** [review]
:question:

File: assembly/__tests__/vm-mock.spec.ts (hunk 4 above)
Comment (Thykof) [reply assembly/__tests__/vm-mock.spec.ts:71]: @AurelienFT ? 

In any case here we are testing the mock so it's not really serious

**Pull request merged**